src/include/uapi/ipc.h

```diff
@@ -497,17 +488,17 @@ struct sof_ipc_host_buffer {
 
 struct sof_ipc_stream_params {
 	struct sof_ipc_host_buffer buffer;
-	enum sof_ipc_stream_direction direction;
-	enum sof_ipc_frame frame_fmt;
-	enum sof_ipc_buffer_format buffer_fmt;
+	uint32_t direction;	/* SOF_IPC_STREAM_ */
+	uint32_t frame_fmt;	/* SOF_IPC_FRAME_ */
+	uint32_t buffer_fmt;	/* SOF_IPC_BUFFER_ */
 	uint32_t stream_tag;
 	uint32_t rate;
 	uint32_t channels;
 	uint32_t sample_valid_bytes;
 	uint32_t sample_container_bytes;
 	/* for notifying host period has completed - 0 means no period IRQ */
 	uint32_t host_period_bytes;
-	enum sof_ipc_chmap chmap[SOF_IPC_MAX_CHANNELS];	/* channel map */
+	uint16_t chmap[SOF_IPC_MAX_CHANNELS];	/* channel map - SOF_CHMAP_ */
 } __attribute__((packed));
```

Comment:
if you use uint16 for channel maps, you could use them as well for direction and other fields. the only one which looks like it needs more than 16 bits is the rate and the period_bytes.

Reply:
uint32_t rate;
uint16_t stream_tag;
uint16_t channels;
uint16_t sample_valid_bytes;
uint16_t sample_container_bytes;